akka-sample-persistence-typed-scala/src/main/scala/sample/persistence/ShoppingCart.scala

```diff
@@ -0,0 +1,119 @@
+package sample.persistence
+
+import akka.actor.typed.ActorRef
+import akka.Done
+import akka.actor.typed.Behavior
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.persistence.typed.scaladsl.Effect
+import akka.actor.testkit.typed.scaladsl.Effects
+import akka.persistence.typed.ExpectingReply
+
+object ShoppingCart {
+    /**
+     * The current state held by the persistent entity.
+    */
+    case class State(items: Map[String, Int], checkedOut: Boolean) {
+        def updateItem(productId: String, quantity: Int) = {
+            quantity match {
+            case 0 => copy(items = items - productId)
+            case _ => copy(items = items + (productId -> quantity))
+            }
+        }
+
+        def checkout = copy(checkedOut = true)
+    }
+    object State {
+        val empty = State(Map.empty, false)
+    }
+    
+
+    /**
+     * This interface defines all the commands that the ShoppingCart persistent actor supports.
+    */
+    sealed trait Command
+
+    sealed trait Result
+    case object OK extends Result
+    case class Failed(msg: String) extends Result
+
+    /**
+     * A command to update an item.
+    *
+    * It can reply with `Done`, which is sent back to the caller when
+    * all the events emitted by this command are successfully persisted.
+    */
+    case class UpdateItem(productId: String, quantity: Int, replyTo: ActorRef[Result]) extends Command with ExpectingReply[Result]
+
+    /**
+     * A command to get the current state of the shopping cart.
+    *
+    * It can reply with the state of the ShoppingCart
+    */
+    case class Get(replyTo: ActorRef[State]) extends Command with ExpectingReply[State]
+
+    /**
+     * A command to checkout the shopping cart.
+    *
+    * It can reply with Done, which will be returned when the fact that checkout
+    * has started has successfully been persisted.
+    */
+    case class Checkout(replyTo: ActorRef[Result]) extends Command with ExpectingReply[Result]
+
+    /**
+     * This interface defines all the events that the ShoppingCart supports.
+     */
+    sealed trait Event
+
+    /**
+     * An event that represents a item updated event.
+     */
+    case class ItemUpdated(productId: String, quantity: Int) extends Event
+
+    /**
+     * An event that represents a checked out event.
+     */
+    case object CheckedOut extends Event
+
+    def behavior(id: String): Behavior[Command] =
+        EventSourcedBehavior[Command, Event, State](
+            PersistenceId(id),
```

Comment:
good convention is to concat with the type name: `s"ShoppingCart|$id"`
That is what is done by `EventSourcedEntity`. We should probably make this a Sharded example as next step, because in almost all cases they are used together.